CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/resourcetypes/CVKImage.java

```diff
@@ -843,81 +817,90 @@ public int SaveMemoryToFile(MemoryStack stack, File file, boolean needsSwizzle)
         VkSubresourceLayout subResourceLayout = VkSubresourceLayout.callocStack(stack);
-        vkGetImageSubresourceLayout(CVKDevice.GetVkDevice(), destImage.GetImageHandle(), subResource, subResourceLayout);
+        vkGetImageSubresourceLayout(CVKDevice.GetVkDevice(), GetImageHandle(), subResource, subResourceLayout);
 
-        // The destination image is in the format VK_FORMAT_B8G8R8A8_UNORM:
+        ////////////////////////////////////////////////////////////////////////
+        // The image is in the format VK_FORMAT_B8G8R8A8_UNORM:
         // CmdBlitImage() automatically converts between formats for us.
         // VK_FORMAT_B8G8R8A8_UNORM
         //    specifies a four-component, 32-bit unsigned normalized format 
         //    that has an 8-bit B component in byte 0, an 8-bit G component 
         //    in byte 1, an 8-bit R component in byte 2, and an 8-bit A component in byte 3.
-
         // The PNG image format is RGB (24-bit)
+        ////////////////////////////////////////////////////////////////////////
 
         // Size (in bytes) of the destination image including any padding
         final long sourceImageSize = subResourceLayout.size(); 
-        // Offset (in bytes) that the image starts (usually 0)
-        long offset = subResourceLayout.offset();
-        // Calculate the file width including the padding (in bytes)
-        int fileWidth = (int)(subResourceLayout.rowPitch() / 4);
-        // Calculate the file height (in bytes)
-        int fileHeight = (int)(sourceImageSize / fileWidth) / 4;
-        // Calculate the file size (in bytes)
-        int fileSize = (int)(sourceImageSize / 4 ) * 3;
-
-        // TODO Hydra need to handle images with layers like the atlas texture
+        int fileWidth = width;
+        int fileHeight = height;
+        int fileSize = (int)sourceImageSize;
+        if (!needsSwizzle) {
+            // Calculate the file width including the padding (in bytes)
+            fileWidth = (int)(subResourceLayout.rowPitch() / 4);
+            CVKAssert(fileWidth != 0);
+            // Calculate the file height (in bytes)
+            fileHeight = (int)(sourceImageSize / fileWidth) / 4;
+            
+            // Calculate the file size (in bytes)
+            //fileSize = (int)(sourceImageSize / 4 ) * 3;
+        }
+        
+        
+        // TODO HYDRA need to handle images with layers like the atlas texture
         long fileLayers = subResourceLayout.depthPitch();
 
         try {
             // Create the output buffered image with the original images
             // width and height, and in the PNG format (3 bytes BGR)
-            out = new BufferedImage(width, height, TYPE_3BYTE_BGR);
+            //out = new BufferedImage(width, height, TYPE_3BYTE_BGR);
+            out = new BufferedImage(width, height, TYPE_INT_ARGB);
+            
+                    
             byte[] rgb = new byte[fileSize];
             int i = 0;
             int r = 0;
```

Comment:
What is r?  Would iSrc and iDst make more sense?

Reply:
yes it would, changed